src/rooms/menu.lua

```diff
@@ -1,226 +1,135 @@
 local pd <const> = playdate
 local sound <const> = pd.sound
 local gfx <const> = pd.graphics
+local ui <const> = pd.ui
+local systemMenu <const> = pd.getSystemMenu()
 
--- Constants / Assets
+class ("Menu").extends(Room)
 
-local spButton = sound.sampleplayer.new("assets/sfx/ButtonSelect")
-local imageSpriteTitle <const> = gfx.image.new("assets/images/title"):invertedImage()
-local imageSpriteRobot <const> = gfx.imagetable.new(assets.imageTables.player)
-local imagetableArrows <const> = gfx.imagetable.new(assets.imageTables.menuArrows)
-
--- Local Variables
-
-local spritesArrows = {}
-local spriteTitle
-local spriteRobot
-local spritePressStart
 local sceneManager
-local fileplayer
-
-local timerTitleAnimation
-local blinkerPressStart
-
--- Level Selection
-
-local indexLevel = nil
-local levels = nil
-
-class("Menu").extends(Room)
+local spButton = sound.sampleplayer.new("assets/sfx/ButtonSelect")
 
-function Menu:init(_levels)
-  levels = _levels
+local CELL_HEIGHT <const> = 110
+local CELL_INSETS <const> = 5
+local CELL_PADDING_V <const> = 8
+local CELL_PADDING_H = 5
+local CELL_WIDTH <const> = 400 - (CELL_INSETS * 2) - (CELL_PADDING_H * 2)
+
+local CELL_FILL_ANIM_SPEED <const> = 800
+
+local gridView = ui.gridview.new(0, CELL_HEIGHT)
```

Comment:
Can we move the gridview implementation to a separate file? Like inside `rooms/menu/gridview.lua`.
That way we can have the settings & function definitions in the same place.

We should also have this initialization code inside a function to avoid it running on game launch. To have external functions while using an initializer, just redirect the function refs like `newGridView.drawCell = localDrawCell`